include/taskolib/SequenceManager.h

```diff
@@ -219,6 +225,102 @@ class SequenceManager
      */
     static UniqueId create_unique_id(const std::vector<SequenceOnDisk>& sequences);
 
+    /**
+     * Perform changes and commit them to the git repository.
+     *
+     * This function allows to introduce some changes to the filesystem and create a
+     * git commit from the changes; but if something fails the repository is rolled
+     * back safely.
+     *
+     * Committed are just changes in the directories given in dirs and the message starts
+     * with the specified message. Both dirs and message can be extended by the actual
+     * actions that are performed.
+     *
+     * - Repository is prepared (reset)
+     * - The actions are performed (closure is executed) - they should change the filesystem
+     * - Git searches for changes (just in the directories given by dirs)
+     * - The changes are added and committed using the given message as title
+     * - The commit body contains details about the changed files
+     * - If at any point an error occurs the commit is aborted and the filesystem is reset to
+     *   where we started (changes are undone)
+     *
+     * Sometimes we do not know the actual path we want change and commit before the real
+     * actions are performed. The closure can return a string; and that is a path to be
+     * also considered/added to the commit. The path as string is also added at the end
+     * of the title.
+     *
+     * The signature of the closure can be `void action()` or `std::string action()`.
+     *
+     * It is not possible to add a path to dirs while keeping the commit title unchanged.
+     *
+     * All dirs are relative to the git repository root.
+     *
+     * \param dirs     List of directories to add for this commit (usually just one)
+     * \param message  The commit message title
+     * \param action   Closure that performs the actual modifications on the filesystem
+     * \returns        true if the commit was successful, false if nothing found to commit
+     * \exception      task::Error, git::Error, or any other type can be thrown
+     *
+     */
+    template <typename T>
+    bool perform_commit(gul14::SmallVector<std::string, 2> dirs, std::string message, T action)
```

Comment:
This looks OK as-is, but we might consider other types:
- Maybe use `std::filesystem::path` instead of `string` just to make it clear that we are dealing with paths? Maybe also as the return type of the closure?
- As much as I love `SmallVector`, I fear it only complicates the signature here without any benefit. Why not simply use `std::vector` instead?

Reply:
If I recollect correctly I had it `path` at some point in time and changed it to strings for some reason.
I will check.

I would rather drop the container completely and have just one path - there is only one case where we need two paths - maybe we can solve that differently. That was exactly the reason why I had in some previous version the function that takes two globs, remember? It all has different downsides :sigh: